command/txpool/txpool_subscribe.go

```diff
@@ -0,0 +1,216 @@
+	t.DefineFlags()
+
+	return helper.GenerateHelp(t.Synopsis(), helper.GenerateUsage(t.GetBaseCommand(), t.FlagMap), t.FlagMap)
+}
+
+// Synopsis implements the cli.Command interface
+func (t *TxPoolSubscribeCommand) Synopsis() string {
+	return t.GetHelperText()
+}
+
+// Run implements the cli.Command interface
+func (t *TxPoolSubscribeCommand) Run(args []string) int {
+	flags := t.Base.NewFlagSet(t.GetBaseCommand(), t.Formatter, t.GRPC)
+
+	var added bool
+	var promoted bool
+	var enqueued bool
+	var dropped bool
+	var demoted bool
+
+	flags.BoolVar(&added, "added", false, "")
+	flags.BoolVar(&promoted, "enqueued", false, "")
+	flags.BoolVar(&enqueued, "enqueued", false, "")
+	flags.BoolVar(&dropped, "dropped", false, "")
+	flags.BoolVar(&demoted, "demoted", false, "")
+
+	if err := flags.Parse(args); err != nil {
+		t.Formatter.OutputError(err)
+		return 1
+	}
+
+	eventTypes := make([]txpoolProto.EventType, 0)
+	if added {
+		eventTypes = append(eventTypes, txpoolProto.EventType_ADDED)
+	}
+
+	if promoted {
+		eventTypes = append(eventTypes, txpoolProto.EventType_PROMOTED)
+	}
+
+	if enqueued {
+		eventTypes = append(eventTypes, txpoolProto.EventType_ENQUEUED)
+	}
+
+	if dropped {
+		eventTypes = append(eventTypes, txpoolProto.EventType_DROPPED)
+	}
+
+	if demoted {
+		eventTypes = append(eventTypes, txpoolProto.EventType_DEMOTED)
+	}
+
+	if len(eventTypes) == 0 {
+		// Any kind of event subscription is default
+		eventTypes = append(
+			eventTypes,
+			txpoolProto.EventType_ADDED,
+			txpoolProto.EventType_PROMOTED,
+			txpoolProto.EventType_ENQUEUED,
+			txpoolProto.EventType_DROPPED,
+			txpoolProto.EventType_DEMOTED,
+		)
+	}
+
+	conn, err := t.GRPC.Conn()
+	if err != nil {
+		t.Formatter.OutputError(err)
+		return 1
+	}
+
+	clt := txpoolProto.NewTxnPoolOperatorClient(conn)
+	ctx, cancelFn := context.WithCancel(context.Background())
+
+	stream, err := clt.Subscribe(ctx, &txpoolProto.SubscribeRequest{
+		Types: eventTypes,
+	})
+	if err != nil {
+		t.Formatter.OutputError(err)
+		cancelFn()
+		return 1
+	}
+
+	doneCh := make(chan struct{})
+	go func() {
+		for {
+			streamEvent, err := stream.Recv()
+			if err == io.EOF {
+				break
+			}
+
+			if err != nil {
+				t.Formatter.OutputError(fmt.Errorf("failed to read event: %w", err))
+				break
+			}
+
+			t.Formatter.OutputResult(&TxPoolEventResult{
+				EventType: streamEvent.Type,
+				TxHash:    streamEvent.TxHash,
+			})
+		}
+		doneCh <- struct{}{}
+	}()
+
+	// wait for the user to quit with ctrl-c
+	signalCh := make(chan os.Signal, 4)
+	signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
+
+	select {
+	case <-signalCh:
+	case <-doneCh:
+	}
+	cancelFn()
+
+	return 0
+}
+
+type TxPoolEventResult struct {
+	EventType txpoolProto.EventType `json:"eventType"`
+	TxHash    string                `json:"txHash"`
+}
+
+func (r *TxPoolEventResult) Output() string {
+	var buffer bytes.Buffer
+
+	// current number & hash
+	buffer.WriteString("\n[TXPOOL EVENT]\n")
```

Comment:
Why are you using a buffer here? You can do:
```
str := ""
str += "\n[TXPOOL EVENT]\n"
...
return str
```
This is how it was done for other command outputs.

Reply:
We actually use a buffer for all command outputs.

Please look at this PR which introduced it:
https://github.com/0xPolygon/polygon-sdk/pull/277